registry/storage/catalog.go

```diff
@@ -49,17 +48,22 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
```

Comment:
Okay, I see why it is useful to not return the `io.EOF` instead of `errFinishedWalk`.  However won't this allow `len(foundRepos) > len(repos)`? In that case an `io.EOF` could be returned, but there are `foundRepos` entries not copied into `repos`, essentially `n < len(foundRepos)`  after `n = copy(repos, foundRepos)`.